### PR TITLE
Installer for VMClarity on Azure

### PR DESCRIPTION
**Pull request by sambetts-cisco**
## Description

Introduces an installer for VMClarity on Azure using Bicep / Azure Resource Manager. VMClarity will be installed into a new Resource Group with the control plane services running in a single VM. A dedicated VMClarity vNet will be created with 2 subnets, one for the control plane services which allows external access to the VMClarity server, and one for the Scanners which is only allows ingress from the control plane network so that it can be used as a bastion.

Two roles are added:
- One for discovery and snapshotting within a subscription.
- One for creating resources within the VMClarity resource group for scanning (VM, Disks, Network Interfaces)

To install on Azure you can either:
- use the az CLI to directly deploy the bicep (vmclarity.bicep)
- manually upload the ARM template (vmclarity.json) to the portal as a template spec
- Click the "Deploy to Azure" button in the README to deploy the ARM template with the custom UI wizard (this is the recommended method)

## Type of Change

[ ] Bug Fix  
[X] New Feature  
[ ] Breaking Change  
[ ] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [X] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [X] Existing issues have been referenced (where applicable)
- [X] I have verified this change is not present in other open pull requests
- [X] Functionality is documented
- [X] All code style checks pass
- [X] New code contribution is covered by automated tests
- [X] All new and existing tests pass

### CODE REVIEW

File: installation/azure/vmclarity.bicep [hunk 1]
```diff
@@ -0,0 +1,151 @@
+targetScope = 'subscription'
+
+@description('Location for the VMClarity resource group')
+param location string = 'eastus'
+
+@description('Username for the Virtual Machine.')
+param adminUsername string
+
+@description('SSH Key for the Virtual Machine. SSH key is recommended.')
```
Comment (fishkerez) [inline installation/azure/vmclarity.bicep:9]: Maybe worth mentioning this is the public key

File: installation/azure/vmclarity.bicep [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+@allowed([
+  'Always'
+  'OnSuccess'
+  'Never'
+])
+param assetScanDeletePolicy string = 'Always'
+
+@description('Database to Use')
+@allowed([
+  //'Postgresql'
```
Comment (fishkerez) [inline installation/azure/vmclarity.bicep:57]: why are those commented out?

Comment (FrimIdan) [reply installation/azure/vmclarity.bicep:57]: Postgres DB driver has issues
https://github.com/openclarity/vmclarity/issues/395
https://github.com/openclarity/vmclarity/pull/416

File: installation/azure/vmclarity.bicep (hunk 1 above)
Comment (sambetts-cisco) [reply installation/azure/vmclarity.bicep]: updated the description

**Pull request merged**